core/observe.c

```diff
@@ -882,6 +872,47 @@ void observe_remove(lwm2m_observation_t * observationP)
     observationP->clientP->observationList = (lwm2m_observation_t *) LWM2M_LIST_RM(observationP->clientP->observationList, observationP->id, NULL);
     lwm2m_free(observationP);
 }
+static
+void prv_applyObservationRequestCallback(observation_data_t * observationData, int status, int block_num, int block_size, bool block_more, lwm2m_media_type_t format, uint8_t * data, int dataLength)
+{
+    if (observationData->blockCallback != NULL)
+    {
+        observationData->blockCallback(observationData->client,
```

Comment:
Calling callbacks without checking arguments is not nice, who knows what the callback does...

Reply:
I think we had similar discussions in other cases and the argument beeing that in order to keep a minimal footprint we need to trust calling functions and callback to not do anything vile.